firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexManager.java

```diff
@@ -84,17 +85,30 @@
   IndexOffset getMinOffset(String collectionGroup);
 
   /**
-   * Returns an index that can be used to serve the provided target. Returns {@code null} if no
-   * index is configured.
+   * Returns an index that can be used to serve the provided target, as well as the number of index
+   * segments that matched the target filters and orderBys. Returns {@code null} if no index is
+   * configured.
    */
   @Nullable
-  FieldIndex getFieldIndex(Target target);
+  Pair<FieldIndex, Integer> getFieldIndexAndSegmentCount(Target target);
```

Comment:
This is probably a personal problem that points to some deeper issue about my adolescence, but I really do not like APIs that return `Pair`. I completely understand why you choose to use a `Pair` here and am not suggesting you should change it, but `Pair`s are very nondescriptive and you essentially had to make up for this with documentation. You could achieve a more concise result by returning a wrapper type (albeit at the cost of code size)

Reply:
FWIW, it also seems like we don't actually use this method in the public API anymore, so we could remove the public declaration and just use `Pair` in SQLiteIndexManager.